### PR TITLE
MAINT: remove dead code in interpolate.py

### PR DESCRIPTION
**Pull request by rgommers**
Those functions were written by Travis and haven't been touched since 2007.

### CODE REVIEW

File: scipy/interpolate/interpolate.py [hunk 1]
```diff
@@ -2633,293 +2625,6 @@ def fromspline(cls, xk, cvals, order, fill=0.0):
...
-    if b.ndim <= 2:
-        return dot(a, b)
-    else:
-        axes = list(range(b.ndim))
-        axes.insert(-1, 0)
-        axes.pop(0)
-        return dot(a, b.transpose(axes))
-
-
-def _find_smoothest(xk, yk, order, B=None):
```
Comment (ev-br) [inline scipy/interpolate/interpolate.py]: Might be too soon remove it yet. This one is used by splmake, which is deprecated but sadly used in the wild (via spline).

Comment (ev-br) [inline scipy/interpolate/interpolate.py]: Other ones (with NotImplementedError) can certainly go, +1 to that.

Comment (rgommers) [inline scipy/interpolate/interpolate.py]: ah, that was very much not obvious. another way ``eval`` can trick you. and no tests:(

**Comment by rgommers** [comment]
should be god to go now

**Comment by ev-br** [comment]
Thanks Ralf, merged.

**Pull request merged**